### PR TITLE
Step three

### PR DESCRIPTION
**Pull request by jCabreraAltimetrik**
Would you mind checking this out ?

### CODE REVIEW

File: force-app/main/default/aura/TaskCardKanban/TaskCardKanbanController.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+({
+	newTask : function(component, event, helper) {
+		let openNewTaskModal = component.getEvent("openNewTaskCard");
+        openNewTaskModal.setParam("typeOfTask",event.getSource().get("v.name"));
+        openNewTaskModal.fire();
+	},
+    completeTask : function(component, event, helper)
+    {
+        let closeTask = component.getEvent("closeTask");
```
Comment (stefaniaoverman) [inline force-app/main/default/aura/TaskCardKanban/TaskCardKanbanController.js]: you should move this logic into a helper. The controller should be something more like an "event handler"

File: force-app/main/default/aura/TaskKanban/TaskKanbanController.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		helper.loadTaskOfContact(component,cId).then(result => {
+            component.set("v.selectedContact",helper.getContactSelectChanged(component,cId));
+            helper.parseContactTaskIntoLists(component,result);
+        }).catch(error => {
+           
+        });         
+	},
+	openNewTaskCardHandler : function(component, event, helper){
+        
+        let cId = component.find("contactSelected");
```
Comment (stefaniaoverman) [inline force-app/main/default/aura/TaskKanban/TaskKanbanController.js]: this kind of logic should be moved into the component's helper.

File: force-app/main/default/classes/TaskKanbanController.cls [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    }
+    @AuraEnabled
+    public static void completeTask(string contactId,string taskId,string state)
+    {
+        List<string> tasksID = new List<string>{taskId};
+        List<Task> tasksToUpdate = new List<Task>();
+        List<Contact> contactsToUpdate = new List<Contact>();
+      /*  for(Task t : [SELECT id,status from Task where id = :taskId])
+        {
+            t.Status = 'Completed';
```
Comment (stefaniaoverman) [inline force-app/main/default/classes/TaskKanbanController.cls]: please remove commented code.

**Review by stefaniaoverman — CHANGES_REQUESTED** [review]

File: force-app/main/default/aura/TaskKanban/TaskKanbanController.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	deleteTaskHandler  : function(component,event,helper)
+	{
+        helper.deleteTaskHandler(component,event).then(result => {
+            helper.displayToast(component,"success","Success!","Task Deleted Successfully!");
+            helper.loadTaskOfContact(component,component.get("v.selectedContact")["Id"]).then(result => {
+                    component.set("v.tasksOfContact",result);
+                    helper.parseContactTaskIntoLists(component,result);
+                    
+                }).catch(error => {
+                    helper.displayToast(component,"error","error!",error);
```
Comment (stefaniaoverman) [inline force-app/main/default/aura/TaskKanban/TaskKanbanController.js]: this is the kind of error messages that Salesforce love 😆

Comment (jCabreraAltimetrik) [reply force-app/main/default/aura/TaskKanban/TaskKanbanController.js]: Agreed ! Added more information on the message of the toast, will add changes them to the next commit

File: force-app/main/default/classes/TaskKanbanController.cls (hunk 3 above)
Comment (jCabreraAltimetrik) [reply force-app/main/default/classes/TaskKanbanController.cls]: Done and done.

File: force-app/main/default/aura/TaskKanban/TaskKanbanController.js (hunk 2 above)
Comment (jCabreraAltimetrik) [reply force-app/main/default/aura/TaskKanban/TaskKanbanController.js]: Yes, Moved all components logic to their respective helper. Will add changes to my next commit.

**Comment by jCabreraAltimetrik** [comment]
- Removed Commented code.
- Changed error messages to add more info.
- Moved components controller methods to components helper

File: force-app/main/default/aura/TaskKanban/TaskKanbanController.js [hunk 5]
```diff
@@ -1,104 +1,30 @@
...
-                helper.loadTaskOfContact(component,component.get("v.selectedContact")["Id"]).then(result => {
-                    component.set("v.tasksOfContact",result);
-                    helper.parseContactTaskIntoLists(component,result);
-                    
-                }).catch(error => {
-                    helper.displayToast(component,"error","error!",error);
-                });  
-            }
-        }
+        helper.processOnContactFromRDLoaded(component,event);
```
Comment (stefaniaoverman) [inline force-app/main/default/aura/TaskKanban/TaskKanbanController.js:8]: this looks good, I like it!

File: force-app/main/default/classes/TaskKanbanController.cls [hunk 6]
```diff
@@ -78,16 +69,8 @@ public without sharing class TaskKanbanController {
...
-                    	
                     t.Status = 'Completed';
-                    if(state == 'w')
-                    {
-                        contactPoints += t.Points__c;
-                    }else 
-                    {
-                        contactPoints -= t.Points__c;
-                    }
+                    contactPoints += (state == 'w') ? t.Points__c : -t.Points__c;
```
Comment (stefaniaoverman) [inline force-app/main/default/classes/TaskKanbanController.cls:73]: nice

**Pull request closed without merging**